### PR TITLE
[hclips] Add new extractor

### PR DESCRIPTION
**Pull request by ekimae-youkai**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Support download video from hclips.com

MP4 URL from HTML is encode with special algorithm which this PR defeat

Example URL: https://www.hclips.com/videos/hottest-homemade-movie-with-milf-brunette-scenes28529/

### CODE REVIEW

File: youtube_dl/extractor/hclips.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor, ExtractorError
+
+
+class HclipsIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?hclips\.com/videos/(?P<id>[a-zA-Z0-9-_]+)/?'
```
Comment (dstftw) [inline youtube_dl/extractor/hclips.py]: `/?` pointless.

File: youtube_dl/extractor/hclips.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            decoded_url += chr((a << 2) | (b >> 4))
+
+            if c != last_char:
+                decoded_url += chr((b & 0xf) << 4 | c >> 2)
+
+            if d != last_char:
+                decoded_url += chr((c & 0x3) << 6 | d)
+
+        if not decoded_url.startswith("http"):
+            raise ExtractorError("Expected URL after decode. Got {}".format(decoded_url))
```
Comment (dstftw) [inline youtube_dl/extractor/hclips.py]: No `{}`.

File: youtube_dl/extractor/hclips.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        return decoded_url
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        video_id = self._html_search_regex(r'https://www.hclips.com/embed/([0-9]*)', webpage, 'id_number', default=display_id)
```
Comment (dstftw) [inline youtube_dl/extractor/hclips.py]: Matching empty string is senseless.

File: youtube_dl/extractor/hclips.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        title = self._html_search_regex(r'<h1>(.+?)</h1>', webpage, 'title', default=display_id)
+
+        encoded_video_url = self._search_regex(r'var video_url="(.*?)"', webpage, 'video_url')
```
Comment (dstftw) [inline youtube_dl/extractor/hclips.py]: Same.

**Comment by afterdelight** [comment]
nice Fara long dick

**Pull request closed without merging**